source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,531 @@
+in the TXT records.
+
+Why explicitly opt-in to this behaviour instead of letting mongos inform the driver of the load balancer?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This was a case of complexity in implementation and the number of affected services, including
+but not limited to mongos, Atlas Proxy, Elastic Front-End, and Data Lake. All services would need
+to be modified to understand the PROXY protocol (although Data Lake already does), or add
+configuration options that are read during start-up. From there they would need to alter their
+hello responses in order to tell the driver they were behind a load balancer.
+
+From a driver's view there is a change in behaviour in that once in load balanced mode, the driver
+can never transition out of that mode, unlike other states in the SDAM specification.
+
+Why does this specification instruct drivers to not check connections back into the connection pool in some circumstances?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the case of a load balancer fronting multiple services, it is possible that a connection to th
+e load balancer could result in a connection behind the load balancer to a different service. In
+order to guarantee these operations execute on the same service they need to be executed on the
+same socket - not checking a connection back into the pool for the entire operation guarantees this.
+
+What reason has a client side connection reaper for idle cursors not been put into this specification?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It was discussed as a potential solution for maxed out connection pools that the drivers could
+potentially behave similar to the server and close long running cursors after a specified time
+period and return their connections to the pool. Due to the high complexity of that solution
+it was determined that better error messaging when the connection pool was maxed out would
+suffice in order for users to easily debug when the pool ran out of connections and fix their
+applications or adjust their pool options accordingly.
+
+Why are we requiring mongos servers to add a new serviceId field in hello responses rather than reusing the existing topologyVersion.processId?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This option was previously discussed, but we opted to add a new :code:`hello` response field in
+order to not mix intentions.
+
+Why does this specification not address load balancer restarts or maintenance?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It was decided that the core problem that was needed to be solved was to introduce durable/resumable
```

Comment:
IIRC, resumable cursors are intended to make cursor iteration more robust because it would tolerate various transient errors. LB maintenance/restarts pose a whole different problem: because there's no monitoring routine, we have no idea if the LB itself is even running and accepting new connections. I'm not sure if there was ever a solution for this problem, but we did throw around the idea of having a "smarter" layer 7 LB that understood the wire protocol and could respond to monitoring requests.

Reply:
I've reworded this paragraph now and split into 2 questions.